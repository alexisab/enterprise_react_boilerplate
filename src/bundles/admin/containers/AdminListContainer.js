import React from 'react';
import { Button } from 'antd';
import AdminForm from '../components/AdminForm';

const AdminListContainer = (props) => {
  const { Dialog } = props;

  return (
    <div>
      <h1>All Admins</h1>
      <Button
        type="primary"
        onClick={(e) => {
          e.preventDefault;
          Dialog.open(
            <AdminForm Dialog={Dialog} />,
            { title: 'Add New Admin' }
          );
        }}>Add Admin</Button>
    </div>
  );
};

export default AdminListContainer;
