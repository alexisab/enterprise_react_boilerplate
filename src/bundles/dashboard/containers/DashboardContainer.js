import React from "react";
import PropTypes from "prop-types";
import Dashboard from "../components/Dashboard";

const DashboardContainer = ({ currentUser }) => (
  <div>
    <Dashboard />
  </div>
);

DashboardContainer.propTypes = {
  currentUser: PropTypes.shape({
    name: PropTypes.string,
    admin: PropTypes.bool,
    super_admin: PropTypes.bool,
    id: PropTypes.string
  }).isRequired
};

export default DashboardContainer;
