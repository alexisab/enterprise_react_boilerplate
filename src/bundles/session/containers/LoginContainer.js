import React from 'react';
import { LoginFormComposer as LoginForm } from '../composers';
// import LoginForm from '../components/LoginForm';
import '../../../css/sessions.css';

const LoginContainer = ({ history }) => (
  <div className="login-page">
    <h1 className="title">Log in</h1>
    <LoginForm history={history} />
  </div>
);

export default LoginContainer;
