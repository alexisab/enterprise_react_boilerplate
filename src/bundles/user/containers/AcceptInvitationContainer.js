import React from 'react';
import { Form, Icon, Input, Button } from 'antd';
import enhance from '../composers/AcceptInvitationComposer';
import '../../../css/sessions.css';

const FormItem = Form.Item;
const AcceptInvitationContainer = ({ history, formData, setFormData, formErrors, handleSave }) => {
  console.log(formErrors);
  return (
    <div className="sign-up">
      <h1 className="title">Accept Invite</h1>
      <div className="signup-form">
        { formErrors.invalid_user &&
          <p className="error-message">{formErrors.invalid_user}</p>
        }
        <Form>
          <FormItem
            validateStatus={formErrors.password && 'error'}
            hasFeedback
            help={formErrors.password}
          >
            <Input
              prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
              name="Password"
              type="password"
              placeholder="Password"
              onChange={e => setFormData({ ...formData, password: e.target.value })}
            />
          </FormItem>
          <FormItem
            validateStatus={formErrors.password_confirmation && 'error'}
            hasFeedback
            help={formErrors.password_confirmation}
          >
            <Input
              prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
              name="Password Confirmation"
              type="password"
              placeholder="Password Confirmation"
              onChange={e => setFormData({ ...formData, password_confirmation: e.target.value })}
            />
          </FormItem>
          <FormItem>
            <Button onClick={(e) => handleSave()} type="primary" htmlType="submit" className="login-form-button">
              Proceed
            </Button>
          </FormItem>
        </Form>
      </div>
    </div>
  );
}

export default enhance(AcceptInvitationContainer);
