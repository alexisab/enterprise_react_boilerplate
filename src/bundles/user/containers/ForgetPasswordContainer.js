import React from 'react';
import { Form, Icon, Input, Button } from 'antd';
import enhance from '../composers/ForgotPasswordComposer';
import '../../../css/sessions.css';

const FormItem = Form.Item;
const AcceptInvitationContainer = ({ history, formData, setFormData, formErrors, handleSave, submit }) => {
  return (
    <div className="sign-up">
      <h1 className="title">Forgot Pasword?</h1>
      <div className="signup-form">
        { submit ?
            <div>
              <p>We have sent the reset password email to you. Please check your mailbox</p>
            </div>
          :
            <div>
              <p>Please key in your email that you signed up below</p>
              <Form>
                <FormItem
                  validateStatus={formErrors.email && 'error'}
                  hasFeedback
                  help={formErrors.email}
                >
                  <Input
                    prefix={<Icon type="mail" style={{ fontSize: 13 }} />}
                    placeholder="Email"
                    name="Email"
                    onChange={e => setFormData({ ...formData, email: e.target.value })}
                  />
                </FormItem>
                <FormItem>
                  <Button onClick={(e) => handleSave()} type="primary" htmlType="submit" className="login-form-button">
                    Send Reset Password Instructions
                  </Button>
                </FormItem>
              </Form>
            </div>
        }

      </div>
    </div>
  );
}

export default enhance(AcceptInvitationContainer);
