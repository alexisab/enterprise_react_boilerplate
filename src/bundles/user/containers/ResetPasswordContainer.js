import React from "react";
import PropTypes from "prop-types";
import { Form, Icon, Input, Button } from "antd";
import enhance from "../composers/ResetPasswordComposer";
import "../../../css/sessions.css";

const FormItem = Form.Item;
const ResetPasswordContainer = ({
  formData,
  setFormData,
  formErrors,
  handleSave
}) => (
  <div className="sign-up">
    <h1 className="title">Reset Password</h1>
    <div className="signup-form">
      {formErrors.invalid_user && (
        <p className="error-message">{formErrors.invalid_user}</p>
      )}
      <Form>
        <FormItem
          validateStatus={formErrors.password && "error"}
          hasFeedback
          help={formErrors.password}
        >
          <Input
            prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
            name="Password"
            type="password"
            placeholder="Password"
            onChange={e =>
              setFormData({ ...formData, password: e.target.value })
            }
          />
        </FormItem>
        <FormItem
          validateStatus={formErrors.password_confirmation && "error"}
          hasFeedback
          help={formErrors.password_confirmation}
        >
          <Input
            prefix={<Icon type="lock" style={{ fontSize: 13 }} />}
            name="Password Confirmation"
            type="password"
            placeholder="Password Confirmation"
            onChange={e =>
              setFormData({
                ...formData,
                password_confirmation: e.target.value
              })
            }
          />
        </FormItem>
        <FormItem>
          <Button
            onClick={() => handleSave()}
            type="primary"
            htmlType="submit"
            className="login-form-button"
          >
            Reset Password
          </Button>
        </FormItem>
      </Form>
    </div>
  </div>
);

ResetPasswordContainer.propTypes = {
  formData: PropTypes.shape({
    password: PropTypes.string,
    password_confirmation: PropTypes.string
  }).isRequired,
  formErrors: PropTypes.shape({
    password: PropTypes.string,
    password_confirmation: PropTypes.string
  }).isRequired,
  handleSave: PropTypes.func.isRequired,
  setFormData: PropTypes.func.isRequired
};

export default enhance(ResetPasswordContainer);
