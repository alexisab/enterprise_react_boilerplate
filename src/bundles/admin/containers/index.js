import AdminListContainer from './AdminListContainer';

export { AdminListContainer }
