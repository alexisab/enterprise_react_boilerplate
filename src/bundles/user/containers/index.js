import SignUpContainer from './SignUpContainer';
import AcceptInvitationContainer from './AcceptInvitationContainer';
import ForgetPasswordContainer from './ForgetPasswordContainer';
import ResetPasswordContainer from './ResetPasswordContainer';

export { SignUpContainer, AcceptInvitationContainer, ForgetPasswordContainer, ResetPasswordContainer };
