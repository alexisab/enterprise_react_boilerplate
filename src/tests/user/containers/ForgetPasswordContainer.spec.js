import React from "react";
import renderer from "react-test-renderer";
import { ForgetPasswordContainer } from "../../../bundles/user/containers/ForgetPasswordContainer";

const history = {
  go: () => {},
  push: () => {}
};

const handleSave = jest.fn();
const setFormData = jest.fn();

const props = {
  handleSave,
  setFormData,
  formData: {
    email: ""
  },
  formErrors: {
    email: ""
  }
};

describe("ForgetPasswordContainer", () => {
  describe("when submit is false", () => {
    test("it should match snapshot", () => {
      const component = renderer.create(
        <ForgetPasswordContainer history={history} submit={false} {...props} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("when submit is true", () => {
    test("it should match snapshot", () => {
      const component = renderer.create(
        <ForgetPasswordContainer history={history} submit {...props} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
