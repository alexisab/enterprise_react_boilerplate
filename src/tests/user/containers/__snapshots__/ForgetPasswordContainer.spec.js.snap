// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgetPasswordContainer when submit is false it should match snapshot 1`] = `
<div
  className="sign-up"
>
  <h1
    className="title"
  >
    Forgot Pasword?
  </h1>
  <div
    className="signup-form"
  >
    <div>
      <p>
        Please key in your email that you signed up below
      </p>
      <form
        className="ant-form ant-form-horizontal"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={undefined}
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span
                  className="ant-input-affix-wrapper"
                  style={undefined}
                >
                  <span
                    className="ant-input-prefix"
                  >
                    <i
                      className="anticon anticon-mail"
                      style={
                        Object {
                          "fontSize": 13,
                        }
                      }
                    />
                  </span>
                  <input
                    className="ant-input"
                    disabled={false}
                    name="Email"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Email"
                    style={null}
                    type="text"
                  />
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={undefined}
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <button
                  className="ant-btn login-form-button ant-btn-primary"
                  onClick={[Function]}
                  type="submit"
                >
                  <span>
                    Send Reset Password Instructions
                  </span>
                </button>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`ForgetPasswordContainer when submit is true it should match snapshot 1`] = `
<div
  className="sign-up"
>
  <h1
    className="title"
  >
    Forgot Pasword?
  </h1>
  <div
    className="signup-form"
  >
    <div>
      <p>
        We have sent the reset password email to you. Please check your mailbox
      </p>
    </div>
  </div>
</div>
`;
